arangod/Replication2/StateMachines/Document/DocumentLeaderState.cpp

```diff
@@ -119,7 +119,7 @@ auto DocumentLeaderState::recoverEntries(std::unique_ptr<EntryIterator> ptr)
       }
     }
 
-    auto doc = DocumentLogEntry{std::string(self->shardId),
+    auto doc = DocumentLogEntry{"",  // this affects all shards
```

Comment:
We should consider making DocumentLogEntry a variant.

Reply:
We agreed to do this in a separate PR.